src/sisl/io/siesta/vibra.py

```diff
@@ -0,0 +1,259 @@
+        self.readline()  # K point
+        self.readline()  # Eigenmode index
+        self.readline()  # Frequency
+        self.readline()  # Eigenmode header
+
+        # Determine number of atoms (i.e. rows per mode)
+        natoms = 0
+        while True:
+            line = self.readline()
+            if line.startswith("Eigenmode"):
+                break
+            else:
+                natoms += 1
+        self._natoms = natoms
+
+        # Determine number of modes
+        nmodes = 1
+        while True:
+            line = self.readline()
+            if line.startswith("k            ="):
+                break
+            elif line.startswith("Eigenvector"):
+                nmodes += 1
+        self._nmodes = nmodes
+
+        # Rewind
+        self.seek(pos)
+
+    def _r_next_eigenmode(self):
+        """Reads the next phonon eigenmode in the vectors file.
+
+        Returns
+        --------
+        EigenmodePhonon:
+            The next eigenmode.
+        """
+        # Skip empty line at the head of the file
+        self.readline()
+
+        k = list(map(float, self.readline().split()[2:]))
+        if len(k) == 0:
+            raise GeneratorExit
+        # Read first eigenvector index
+
+        # Determine number of atoms (i.e. rows per mode)
+        state = np.empty((self._nmodes, self._natoms, 3), dtype=np.complex128)
+        c = np.empty(self._nmodes, dtype=np.float64)
+        for imode in range(self._nmodes):
+            self.readline()
+
+            # Read frequency
+            c[imode] = float(self.readline().split("=")[1])
+
+            # Read real part of eigenmode
+            # Skip eigenmode header
+            self.readline()
+            for iatom in range(self._natoms):
+                line = self.readline()
+                state[imode, iatom, :].real = list(map(float, line.split()))
+
+            # Read imaginary part of eigenmode
+            # Skip eigenmode header
+            self.readline()
+            for iatom in range(self._natoms):
+                line = self.readline()
+                state[imode, iatom, :].imag = list(map(float, line.split()))
+
+        info = dict(k=self._convert_k(k), parent=self.parent)
+        return EigenmodePhonon(state, c * _hc_eV, **info)
+
+    def yield_eigenmode(self):
+        """Iterates over the states in the vectors file
+
+        Yields
+        ------
+        EigenmodePhonon
+        """
+        # Open file and get parsing information
+        self._setup_parsing(close=False)
+
+        try:
+            # Iterate over all eigenstates in the WFSX file, yielding control to the caller at
+            # each iteration.
+            while True:
+                yield self._r_next_eigenmode()
+            # We ran out of eigenstates
+        except GeneratorExit:
+            # The loop in which the generator was used has been broken.
+            pass
+
+    def read_eigenmode(self, k=(0, 0, 0), ktol=1e-4) -> EigenmodePhonon:
```

Comment:
could you use typing for `ktol` -> `ktol: float = 1e-4`, then you may delete the `float, optional` in the doc-str